### PR TITLE
8257639: Update usage of "type" terminology in java.lang.Enum & java.lang.Record

### PR DESCRIPTION
**Pull request by FrauBoes**
This change applies a stricter semantic distinction of 'type' versus 'class and interface'. This is based on the JLS changes described in the "Consistent Class and Interface Terminology" document: https://download.java.net/java/early_access/jdk16/docs/specs/class-terminology-jls.html.

The following rules were applied:
- 'class' and/or 'interface' are used when referring to the class/interface itself
- 'type' is used when referring to the type of a variable or expression

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8257639](https://bugs.openjdk.java.net/browse/JDK-8257639): Update usage of "type" terminology in java.lang.Enum & java.lang.Record

### Reviewers
 * [Chris Hegarty](https://openjdk.java.net/census#chegar) (@ChrisHegarty - **Reviewer**)
 * [Daniel Fuchs](https://openjdk.java.net/census#dfuchs) (@dfuch - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1670/head:pull/1670`
`$ git checkout pull/1670`

### CODE REVIEW

File: src/java.base/share/classes/java/lang/Enum.java [hunk 1]
```diff
@@ -59,7 +59,7 @@
  * {@linkplain java.util.EnumSet set} and {@linkplain
  * java.util.EnumMap map} implementations are available.
  *
- * @param <E> The enum type subclass
+ * @param <E> The enum class subclass
```
Comment (dfuch) [inline src/java.base/share/classes/java/lang/Enum.java:62]: I wonder about this one, given that `<E>` is a type. It sounds strange to me - even though I understand that what is meant is that this type should point to a class (a subclass of `java.lang.Enum`).

Comment (FrauBoes) [reply src/java.base/share/classes/java/lang/Enum.java:62]: Makes sense. I suggest the following instead: `The type of enum subclass`.

Comment (dfuch) [reply src/java.base/share/classes/java/lang/Enum.java:62]: Maybe: "`The type of the enum subclass.`" would sound better - or just "`The enum subclass`".

**Review by ChrisHegarty — APPROVED** [review]

**Review by dfuch — APPROVED** [review]

**Pull request closed without merging**